core/app/models/spree/order.rb

```diff
@@ -158,6 +158,8 @@ def states
     scope :complete, -> { where.not(completed_at: nil) }
     scope :incomplete, -> { where(completed_at: nil) }
     scope :not_canceled, -> { where.not(state: 'canceled') }
+    scope :with_deleted_bill_address, -> { where(bill_address_id: Address.deleted) }
```

Comment:
```suggestion
    scope :with_deleted_bill_address, -> { joins(:bill_address).where.not(Address.table_name => { deleted_at: nil }) }
```

this query is much faster